### PR TITLE
Some sloppy 1.18[.2] work

### PR DESCRIPTION
**Pull request by quat1024**
* Fix Spotless and PMD; update code styles to the newer conventions used in Botania
  * Also run `spotlessApply` and fix [these](https://github.com/quat1024/Psi/actions/runs/2403480276) PMD warnings
* Fix Github Actions CI - code style is checked and artifacts uploaded
* Clean up Gradle buildscript a little bit (mostly bringing it in line with botania and quark)
  * `incrementBuildNumber` got lost along the way, sorry
* Very rough 1.18.2 work. Dependencies updated to 1.18.2 versions. Compiles and runs just fine.

Idk if these also happened in 1.18.1:
* Spell programmer is buggy and often doesn't let you complete spells (to be fair it did this before i touched it as well)
* Exosuit renders fine, but leggings spells don't work (as far as i can tell with the broken programmer); possibly others too
* Graphical glitching (translation stateleak) whenever a spell bullet icon is rendered above the hotbar

Sending a PR mainly in the interest of not doing this
image

### CODE REVIEW

File: src/main/java/vazkii/psi/api/PsiAPI.java [hunk 1]
```diff
@@ -60,28 +58,23 @@ public final class PsiAPI {
...
-	public static Capability<ISocketable> SOCKETABLE_CAPABILITY = CapabilityManager.get(new CapabilityToken<>() {
-	});
+	public static Capability<ISocketable> SOCKETABLE_CAPABILITY = CapabilityManager.get(new CapabilityToken<>() {});
 
 	public static final String MOD_ID = "psi";
 
 	public static final ResourceKey<Registry<Class<? extends SpellPiece>>> SPELL_PIECE_REGISTRY_TYPE_KEY = Registry.createRegistryKey("spell_piece_registry_type_key");
-	private static final MappedRegistry<Class<? extends SpellPiece>> spellPieceRegistry = (MappedRegistry<Class<? extends SpellPiece>>) Registry.registerSimple(SPELL_PIECE_REGISTRY_TYPE_KEY, Lifecycle.stable(), () -> PieceTrickDebug.class);
+	//private static final MappedRegistry<Class<? extends SpellPiece>> spellPieceRegistry = (MappedRegistry<Class<? extends SpellPiece>>) Registry.registerSimple(SPELL_PIECE_REGISTRY_TYPE_KEY, Lifecycle.stable(), () -> PieceTrickDebug.class);
+	private static final MappedRegistry<Class<? extends SpellPiece>> spellPieceRegistry = new MappedRegistry<>(SPELL_PIECE_REGISTRY_TYPE_KEY, Lifecycle.stable(), null); //TODO (circa 1.18.2): un-duct-tape this
```
Comment (quat1024) [inline src/main/java/vazkii/psi/api/PsiAPI.java:77]: This is something to look at; registerSimple became private in 1.18.2 and I'm not sure what the replacement is (for creating synced registries). I'm also not convinced that a vanilla registry is really needed here, though; but it made its way into public Psi API

File: src/main/java/vazkii/psi/api/cad/EnumCADComponent.java [hunk 2]
```diff
@@ -22,7 +24,7 @@ public enum EnumCADComponent {
 	DYE;
 
 	public String getName() {
-		return "psi.component." + name().toLowerCase();
+		return "psi.component." + name().toLowerCase(Locale.ROOT);
```
Comment (quat1024) [inline src/main/java/vazkii/psi/api/cad/EnumCADComponent.java:27]: Fixing PMD helped me catch like 15 of these.

File: src/main/java/vazkii/psi/client/core/handler/ShaderHandler.java [hunk 3]
```diff
@@ -161,56 +160,12 @@ private static String getLogInfo(int obj) {
...
-		} finally {
-			try {
-				in.close();
-			} catch (Exception exc) {
-				if (exception == null) {
-					exception = exc;
-				} else {
-					exc.printStackTrace();
-				}
+		try (InputStream in = ShaderHandler.class.getResourceAsStream(filename)) {
```
Comment (quat1024) [inline src/main/java/vazkii/psi/client/core/handler/ShaderHandler.java:163]: using a try-with-resources here was also a PMD suggestion, nooooo idea what the hell was going on before

File: src/main/java/vazkii/psi/common/block/base/ModBlocks.java [hunk 4]
```diff
@@ -41,40 +38,43 @@
...
-	public static final Block psimetalPlateWhite = new Block(Block.Properties.of(Material.METAL).strength(5, 10).sound(SoundType.METAL));
-	public static final Block psimetalPlateWhiteLight = new Block(Block.Properties.of(Material.METAL).strength(5, 10).sound(SoundType.METAL).lightLevel((blockstate) -> 15));
-	public static final Block psimetalEbony = new Block(Block.Properties.of(Material.METAL).strength(5, 10).sound(SoundType.METAL));
-	public static final Block psimetalIvory = new Block(Block.Properties.of(Material.METAL).strength(5, 10).sound(SoundType.METAL));
-	public static final MenuType<ContainerCADAssembler> containerCADAssembler = IForgeMenuType.create(ContainerCADAssembler::fromNetwork);
-
-	public static final BlockEntityType<TileCADAssembler> ASSEMBLER = createBlockEntityType(TileCADAssembler::new, cadAssembler);
-	public static final BlockEntityType<TileProgrammer> PROGRAMMER = createBlockEntityType(TileProgrammer::new, programmer);
-	public static final BlockEntityType<TileConjured> CONJURED = createBlockEntityType(TileConjured::new, conjured);
+	public static Block cadAssembler;
```
Comment (quat1024) [inline src/main/java/vazkii/psi/common/block/base/ModBlocks.java:48]: This is how I fixed "registry has already been frozen", I'm sure there is a nicer way to do it

File: src/main/java/vazkii/psi/api/PsiAPI.java (hunk 1 above)
Comment (gdavid04) [reply src/main/java/vazkii/psi/api/PsiAPI.java:77]: Using an access transformer would be one solution to registerSimple being private.

Comment (quat1024) [reply src/main/java/vazkii/psi/api/PsiAPI.java:77]: i could, but i'm not sure if it's been made private for a reason; a lot of odd things happened to registries in 1.18.2 and i think forge has some custom registry stuff that i'm not familiar with

**Comment by Floofle** [comment]
I have another bug to report: trying to cast circle bullet anything crashes the game, 
https://pastebin.com/SNG2Phi2

No other mods installed. 

Also you seem to already know about this but the config for tricks and whatnot doesn't let you click the arrow about half the time unless you close and reopen the programmer.

**Comment by quat1024** [comment]
I'm not sure what's up with the buttons on the programmer but I think it's
something like:

* if you select something with a "Position" property, it creates four arrow
buttons that are designated to set the "Position" property on the
highlighted element
* if you select something else without a "position" property, it re-uses
those buttons instead of removing and re-adding them, so they don't work

**Comment by Floofle** [comment]
That sounds consistent with what I was experiencing, yeah. 

Let's say you place a Trick: Explode, which wants a position and a number. If you click a direction for the number (say, up) it won't let you select any of the 4 arrows for position until you close and reopen. If you open a different piece, the same selectors don't work either. 

So any attempt after the first to select the config options breaks, and I don't know enough about programming to guess at the why.

**Pull request merged**